packages/dom/src/clone-dom.js

```diff
@@ -26,6 +26,14 @@ export function cloneNodeAndShadow({ dom, disableShadowDOM }) {
 
     parent.appendChild(clone);
 
+    // shalow clone should not contain children
+    // these might get added due to DOM side effects
+    if (clone.children) {
+      for (const child of Array.from(clone.children)) {
+        clone.removeChild(child);
```

Comment:
clone.children is already an array I believe.

Use this may be

clone.children.forEach(clone.removeChild)

Reply:
`element.children` returns a [HTMLCollection](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection) type ([ref](https://developer.mozilla.org/en-US/docs/Web/API/Element/children)), which is not iterable by default. Hence we need this

Reply:
okay, lets do it like this? `Array.from(clone.children).forEach(clone.removeChild)`

Reply:
<img width="375" alt="Screenshot 2023-04-07 at 11 26 39 PM" src="https://user-images.githubusercontent.com/2987087/230655166-94141b7a-b437-4a71-9a51-d6d0790145a7.png">

Reply:
oops yup syntax, my bad